johnnycanencrypt/__init__.py

```diff
@@ -408,6 +410,70 @@ def add_userid(self, key: Key, userid: str, password: str) -> Key:
         # Regnerate the key object and return it
         return self.get_key(fingerprint)
 
+    def revoke_userid(self, key: Key, userid: str, password: str) -> Key:
+        """Revokes the given user id to the given key, saves on the database. Then returns the modified key object
+
+        :param key: The secret key object
+        :param uid: The string value to add the keybobject
+        :param password: The password for the secret key
+
+        :returns: Key object
+        """
+        if key.keytype != KeyType.SECRET:
+            raise ValueError(f"The {key} is not a secret key.")
+        # A list of UID values which is already in the database
+        olduids = [uid["value"] for uid in key.uids]
```

Comment:
I don't think olduids is needed in this function since you are just setting the userid's revoke column in SQL instead of comparing with olduids like you do in new uid addition.